ruby/ext/google/protobuf_c/encode_decode.c

```diff
@@ -645,21 +651,19 @@ VALUE Message_decode(VALUE klass, VALUE data) {
 
   const upb_pbdecodermethod* method = msgdef_decodermethod(desc);
   const upb_handlers* h = upb_pbdecodermethod_desthandlers(method);
-  upb_pbdecoder decoder;
-  upb_sink sink;
-  upb_status status = UPB_STATUS_INIT;
+  upb_env e;
+  upb_env_init(&e);
```

Comment:
Do we have a benchmark for Ruby? I ask because as written, this would be a performance regression, since it's allocating the encoder/decoders with malloc instead of on the stack. But I also know Ruby imposes a lot of overhead, so I'm just wondering if this registers at all.